backend/services/api_monetization.py

```
@@ -340,3 +340,24 @@ async def record_api_usage(self, api_key: str, endpoint: str, method: str,
+            status_code=status_code,
+            request_size_bytes=0,
+            response_size_bytes=0,
+            cost=Decimal('0.00'),
+            metadata={}
+        )
+
+        # Calculate cost (best-effort). If pricing calculation fails, continue and record with zero cost.
+        try:
+            cost = await self.pricing_calculator.calculate_cost(usage_record, tier)
+            usage_record.cost = cost
+        except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)